jwql/utils/utils.py

```diff
@@ -117,3 +122,44 @@ def filename_parser(filename):
         raise ValueError('Provided file {} does not follow JWST naming conventions (jw<PPPPP><OOO><VVV>_<GGSAA>_<EEEEE>_<detector>_<suffix>.fits)'.format(filename))
 
     return filename_dict
+
+def fileroot_parser(fileroot):
```

Comment:
It is almost the same as `filename_parser` except it doesn't have the suffix and file extension included in the regex search - so that you can still get proposal/detector information/etc. from a string like `jw86600006001_02101_00008_guider1`rather than needing it to be `jw86600006001_02101_00008_guider1_cal.fits`. Do you think it's necessary?

If so, I'm thinking now it might make more sense to include it as another case within `filename_parser` rather than its own function...

Reply:
@laurenmarietta Yes, I think we should try making `filename_parser` more robust instead of having a second, mostly-similar function (which goes against the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)).  I'm thinking it should be possible by adding some conditions on how one builds the string that is passed to `re.compile()` based on if the filename has `.fits` at the end, for example.

Perhaps we could also have a hard coded `FILETYPE_LIST` or something in `utils` that stores all of the valid filetypes (i.e. `uncal`, `cal`, `rate`, etc.).  It could also come in handy in this function.